### ISSUE TITLE
NPM doesn't install when using precise

### ISSUE THREAD

**Issue by vkurup** [caktus/django-project-template#138]
There's another regression in salt 2014.7. A bug in the `pkgrepo` state causes it to always return `True` when testing for the presence of a PPA, so the nodejs PPA never gets installed, and npm is included only in the newer versions of nodejs available via the PPA. This is not an issue on ubuntu 14.04, because the version in the standard repo there is new enough to include npm.

Bug: https://github.com/saltstack/salt/issues/17709

This is apparently fixed on master, so hopefully will be fixed whenever that's released...

**Comment by daaray** [comment]
Is the workaround, at present, anything more than just logging onto the box and manually `sudo apt-get install npm`?

**Comment by vkurup** [comment]
I've been doing this (manually on the server):
```
sudo add-apt-repository ppa:chris-lea/node.js 
sudo apt-get update
sudo apt-get install nodejs
```
That version of `nodejs` includes the `npm` command.

A better workaround would be to do the first command in a `cmd.run` state, but I kinda assumed that this would have been fixed in salt by now.

**Comment by daaray** [comment]
:+1:

**Comment by ironfroggy** [comment]
+1

**Comment by mlavin** [comment]
This should be fixed in `2014.7.1` which was released earlier this month https://github.com/saltstack/salt/commit/447d69661dc8ff5edbdc75e901010af28d8c2994

**Comment by vkurup** [comment]
Yes, just confirmed that this works again in 2014.7.1.
